draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1072,328 @@ The following describes each member of the `concise-bom-tag` map.
+>
+> * How Reference Values are conveyed (e.g., using Reference Value triples, or CoSWID tags)
+> * How Reference Values are associated with the Target Environment
+> * How the Target Environment is identified in Evidence
+> * How measurements from Evidence are matched against Reference Values
+
+> A CoRIM profile MUST describe:
+>
+> * How Endorsed Values are conveyed (e.g., using Endorsed Value triples, or Device Identity triples)
+> * How Endorsed Values are added to the Accepted Claims Map
+
+The sections below describe the algorithm used to compare CoMID tags.
+
+### Comparing and processing CoMID tags
+
+There can be multiple passes of the CoMID evidence appraisal process.
+
+In each pass, each potential CoMID tag in the Appraisal Context is compared
+against the accepted claims set (which has been initialised with Evidence from
+the attestor(s) ).
+If the reference values in the CoMID tag match the Accepted Claims Set then
+Endorsements and other values from the tag are added to the Accepted Claims Set.
+
+If any tags matched in a pass through the CoMID appraisal process then another
+pass is performed.
+Tags which matched in an earlier pass are ignored in later passes of the same
+appraisal.
+
+A later pass may match Reference Values from the CoMID against Endorsements
+added by matched tags in an earlier pass, this allows CoRIM authors to create
+hierarchical descriptions of a system.
+
+### Matching Evidence against Reference Values
+
+This section describes the process performed by the verifier to determine
+whether a candidate CoMID tag matches the Accepted Claims Set.
+If a tag does not match the Accepted Claims Set then it is silently ignored
+for this pass, the Verifier continues to iterate over the remaining candidate
+tags.
+
+The Verifier iterates over the Reference Values in the CoMID tag.
+If the Accepted Claims Set does not contain an entry with the same `environment-map`
+as the Reference Value then the tag does not match.
+Comparison of `environment-map` is performed using a binary comparison.
+A Verifier SHOULD convert `environment-map` to the canonical format before
+performing the binary comparison.
+
+The Verifer locates the `measurement-values-map` corresponding to the
+`environment-map` in the Reference Value and the Accepted Claims Set.
+The Verifier enumerates over the key-value pairs in the Reference Value
+`measurement-values-map`.
+If the Accepted Claims Set `measurement-values-map` does not have an entry
+with the same key as the Reference Value `measurement-values-map`,
+or if the entries do not match, then the tag does not match.
+
+The algorithm used to match the `measurement-values-map` entries together
+depends on whether the reference value is tagged,
+and on the `measurement-values-map` key which identifies the entry.
+
+If the Reference Value `measurement-values-map` value starts with a CBOR tag
+then the verifier MUST use the algorithm associated with that tag to match
+the entries.
```

Comment:
CBOR tags could have multiple meanings: (a) data format, (b) function that is applied to the data.  I don't know if we have defined cddl that implies (b) but if we do, we should discuss the implications of this. In the case of `min-svn` there is an expectation that a 'min' function is applied when evaluating an `svn` with a `min-svn`.  However, a tagged-min-svn only indirectly implies the 'min' function semantics as the data type itself implies function semantics. 
It is reasonable to expect that function semantics could be explicitly described by the cddl expression contained in the 'value' side of a 'key-value' expression. The `min-svn` example encodes the function semantics in the 'key' rather than the 'value'.